packages/animations/browser/src/render/shared.ts

```diff
@@ -124,19 +124,10 @@ export function makeAnimationEvent(
   return {element, triggerName, fromState, toState, phaseName, totalTime, disabled: !!disabled};
 }
 
-export function getOrSetAsInMap(
-    map: Map<any, any>|{[key: string]: any}, key: any, defaultValue: any) {
-  let value: any;
-  if (map instanceof Map) {
-    value = map.get(key);
-    if (!value) {
-      map.set(key, value = defaultValue);
-    }
-  } else {
-    value = map[key];
-    if (!value) {
-      value = map[key] = defaultValue;
-    }
+export function getOrSetAsInMap<T, V>(map: Map<T, V>, key: T, defaultValue: V) {
+  let value = map.get(key);
+  if (!value) {
+    map.set(key, value = defaultValue);
```

Comment:
It looks like all call sites right now pass `new Map()` as a default value. We could probably avoid doing this and instead move the code inside the function itself (also a perf benefit here: we'd not create an instance of a `Map` when we don't really need it).

```suggestion
export function getOrSetAsInMap<T, V>(map: Map<T, V>, key: T): V {
  let value = map.get(key);
  if (!value) {
    map.set(key, value = new Map());
```

Reply:
We could probably rename a function as well (not sure what'd be a good name though :)).

Reply:
Yeah I'm not a fan of the name. I'll play around and see if I can come up with something better.

Reply:
> It looks like all call sites right now pass `new Map()` as a default value. We could probably avoid doing this and instead move the code inside the function itself (also a perf benefit here: we'd not create an instance of a `Map` when we don't really need it).

There is still one location where this is not the case. I'll see what I can do to address it.

Reply:
As it turns out, there's a number of situations where the default value is not a Map. There are usages where it's an array, a Set, or a Map. So this needs to be as is for now.